From akpm@linux-foundation.org Mon Feb  5 16:42:42 2007
From: Zheng XiaoJun <zhengxiaojun@hsic.com.cn>
Date: Mon, 05 Feb 2007 16:40:57 -0800
Subject: USB: input: HID: add CIDC USB device to HID blacklist
To: greg@kroah.com
Cc: akpm@linux-foundation.org, zhengxiaojun@hsic.com.cn, dtor@mail.ru
Message-ID: <200702060042.l160g10Q005239@shell0.pdx.osdl.net>


From: Zheng XiaoJun <zhengxiaojun@hsic.com.cn>

Add CIDC USB device to HID blacklist since it is actually a USB token
and has its own driver.

Signed-off-by: Zheng XiaoJun <zhengxiaojun@hsic.com.cn>
Cc: Dmitry Torokhov <dtor@mail.ru>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/input/hid-core.c |    3 +++
 1 file changed, 3 insertions(+)

--- gregkh-2.6.orig/drivers/usb/input/hid-core.c
+++ gregkh-2.6/drivers/usb/input/hid-core.c
@@ -515,6 +515,7 @@ void usbhid_close(struct hid_device *hid
 
 #define USB_VENDOR_ID_TURBOX		0x062a
 #define USB_DEVICE_ID_TURBOX_KEYBOARD	0x0201
+#define USB_VENDOR_ID_CIDC		0x1677
 
 /*
  * Initialize all reports
@@ -953,6 +954,8 @@ static const struct hid_blacklist {
 
 	{ USB_VENDOR_ID_SONY, USB_DEVICE_ID_SONY_PS3_CONTROLLER, HID_QUIRK_SONY_PS3_CONTROLLER },
 
+	{ USB_VENDOR_ID_CIDC, 0x0103, HID_QUIRK_IGNORE },
+
 	{ 0, 0 }
 };
 
